lib/linguist/languages.yml

```diff
@@ -1577,7 +1577,7 @@ HCL:
   ace_mode: ruby
   codemirror_mode: ruby
   codemirror_mime_type: text/x-ruby
-  tm_scope: source.ruby
+  tm_scope: source.terraform
```

Comment:
Scope names usually follow the pattern `source.language_name` but that's totally up to you.
I'm not sure though, is Terraform the name of the language or the name of a software?